src/components/books.js

```diff
@@ -1,16 +1,16 @@
 import React from 'react';
+import { useSelector } from 'react-redux';
 import Book from './book';
 import Form from './form';
 
 const Books = () => {
-  const books = [{ title: 'goodbook', author: 'ekene', id: 1 },
-    { title: 'Collins', author: 'emeka', id: 2 },
-  ];
+  const displayData = useSelector((state) => state.reducers);
+
```

Comment:
- [ ] <b>OPTIONAL:</b> It would also be nice to use `shallowEqual` to make sure that you only re-render when it is necessary. This is because useSelector will always force your component to re-render every time the state changes, and to avoid that, you must make sure that useSelector will re-render the component if and only if the previous books are different from the new books in the state(change here can be even one property on one of the book objects), and since the arrays and objects are not primitive values, you can't use referential equality(`===` Because it will always return false). so, that is where  `shallowEqual` comes to the rescue. 

```jsx
import { useSelector, shallowEqual } from 'react-redux';

// And the you use it like this
const displayData = useSelector((state) => state.reducers, shallowEqual);
```